index.js

```diff
@@ -142,54 +156,169 @@ function updateSchemesAndHost(req) {
   }
 }
 
-module.exports.init = (aApp, aPredefinedSpec, aPath, aWriteInterval, aApiDocsPath = 'api-docs') => {
-  app = aApp;
-  predefinedSpec = aPredefinedSpec;
-  const writeInterval = aWriteInterval || 10 * 1000;
+/** TODO - type defs for Express don't work (I tried @external) */
+/**
+ * Apply this **first**!
+ *
+ * (straight after creating the express app (as the very first middleware))
+ *
+ * @description `response` middleware.
+ *
+ * @param {Express} expressApp - the express app
+ *
+ * @param {Object} [options] optional configuration options
+ * @param {string|undefined} [options.pathToOutputFile=undefined] where to write the openAPI specification to.
+ * Specify this to create the openAPI specification file.
+ * @param {number} [options.writeIntervalMs=10000] how often to write the openAPI specification to file
+ *
+ * @returns void
+ */
+function injectResponseMiddleware(expressApp, options = { pathToOutputFile: undefined, writeIntervalMs: 1000 * 10 }) {
+  /**
+   * save the `expressApp` to our local `app` variable.
+   * Used here, but not in `injestRequestMiddleware`,
+   * because this comes before it.
+   */
+  app = expressApp;
+
+  const { pathToOutputFile, writeIntervalMs } = options;
 
-  // middleware to handle responses
+  /** middleware to handle RESPONSES */
+  // eslint-disable-next-line complexity
   app.use((req, res, next) => {
     try {
       const methodAndPathKey = getMethod(req);
+
       if (methodAndPathKey && methodAndPathKey.method) {
         processors.processResponse(res, methodAndPathKey.method);
       }
-      let firstTime = true;
+
+      let firstTime = true; /** run instantly the first time. TODO do not set set `lastRecordTime` at the start until we run */
       const ts = new Date().getTime();
-      if (firstTime || aPath && ts - lastRecordTime > writeInterval) {
+
+      if (firstTime || pathToOutputFile && ts - lastRecordTime > writeIntervalMs) {
         firstTime = false;
         lastRecordTime = ts;
-        fs.writeFile(aPath, JSON.stringify(spec, null, 2), 'utf8', err => {
-          const fullPath = path.resolve(aPath);
+
+        fs.writeFile(pathToOutputFile, JSON.stringify(spec, null, 2), 'utf8', err => {
+          const fullPath = path.resolve(pathToOutputFile);
+
           if (err) {
             throw new Error(`Cannot store the specification into ${fullPath} because of ${err.message}`);
           }
         });
       }
-    } catch (e) {}
-    next();
+    } catch (e) {
+      /** TODO - shouldn't we do something here? */
```

Comment:
I have a bug here: in line 207 I throw exception and here catch it without logging it out. It is ok to leave it as it is, just to highlight the problem.

Reply:
I think we should fix the bug here.
Should we re-throw (and/or log) the error in the `catch (e) {}` block?

Reply:
I would leave it as it is for now because:
* it is a bit wrong just to log it to console - we can assume that service which uses this module will use some logging module (https://www.npmjs.com/package/bunyan for example) and thus logging becomes inconsistant
* this error doesn't seem too important to interrup the execution of the service even in dev. env.

Reply:
Alright, I'll add a note to the code about this.